### PR TITLE
element-desktop: 1.11.87 -> 1.11.89, element-{web,desktop}: move to pkgs/by-name, format

### PR DESCRIPTION
**Pull request by teutat3s**
https://github.com/element-hq/element-desktop/releases/tag/v1.11.88
https://github.com/element-hq/element-desktop/releases/tag/v1.11.89

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by teutat3s** [comment]
There seem to be [some problems](https://github.com/element-hq/element-web/releases/tag/v1.11.89) with this version, marking WIP for now.

File: pkgs/by-name/el/element-desktop/package.nix [hunk 1]
```diff
@@ -13,21 +13,19 @@
   element-web,
   sqlcipher,
   callPackage,
-  Security,
-  AppKit,
-  CoreServices,
+  darwin,
```
Comment (khaneliman) [inline pkgs/by-name/el/element-desktop/package.nix:16]:
```suggestion
```

File: pkgs/by-name/el/element-desktop/package.nix [hunk 2]
```diff
@@ -13,21 +13,19 @@
...
 }:
 
 let
-  pinData = import ./pin.nix;
+  pinData = import ./element-desktop-pin.nix;
   inherit (pinData.hashes) desktopSrcHash desktopYarnHash;
   executableName = "element-desktop";
-  keytar = callPackage ./keytar { inherit Security AppKit; };
-  seshat = callPackage ./seshat { inherit CoreServices; };
+  keytar = callPackage ./keytar { inherit (darwin.apple_sdk.frameworks) Security AppKit; };
```
Comment (khaneliman) [inline pkgs/by-name/el/element-desktop/package.nix:27]:
```suggestion
  keytar = callPackage ./keytar {  };
```

File: pkgs/by-name/el/element-desktop/package.nix [hunk 3]
```diff
@@ -13,21 +13,19 @@
...
+  seshat = callPackage ./seshat { inherit (darwin.apple_sdk.frameworks) CoreServices; };
```
Comment (khaneliman) [inline pkgs/by-name/el/element-desktop/package.nix:28]:
```suggestion
  seshat = callPackage ./seshat {  };
```

**Review by khaneliman — COMMENTED** [review]
Don't need to include darwin frameworks anymore, they are in stdenv.

**Comment by khaneliman** [comment]
Sorry, also need to remove it from inputs in those files.

File: pkgs/by-name/el/element-desktop/keytar/default.nix [hunk 4]
```diff
@@ -39,8 +37,8 @@ stdenv.mkDerivation rec {
   buildInputs =
     lib.optionals (!stdenv.hostPlatform.isDarwin) [ libsecret ]
     ++ lib.optionals stdenv.hostPlatform.isDarwin [
-      Security
-      AppKit
+      stdenv.Security
+      stdenv.AppKit
     ];
```
Comment (khaneliman) [inline pkgs/by-name/el/element-desktop/keytar/default.nix:42]: You can just remove them here, too.

File: pkgs/by-name/el/element-desktop/seshat/default.nix [hunk 5]
```diff
@@ -37,7 +36,7 @@ rustPlatform.buildRustPackage rec {
     yarn
     fixup-yarn-lock
   ];
-  buildInputs = [ sqlcipher ] ++ lib.optional stdenv.hostPlatform.isDarwin CoreServices;
+  buildInputs = [ sqlcipher ] ++ lib.optional stdenv.hostPlatform.isDarwin stdenv.CoreServices;
```
Comment (khaneliman) [inline pkgs/by-name/el/element-desktop/seshat/default.nix:39]:
```suggestion
  buildInputs = [ sqlcipher ];
```

File: pkgs/by-name/el/element-desktop/keytar/default.nix (hunk 4 above)
Comment (teutat3s) [reply pkgs/by-name/el/element-desktop/keytar/default.nix:42]: Aaah, thanks.

File: pkgs/by-name/el/element-desktop/seshat/default.nix (hunk 5 above)
Comment (teutat3s) [reply pkgs/by-name/el/element-desktop/seshat/default.nix]: Should be done right now.

**Comment by khaneliman** [comment]
Looks like something just needs to be formatted, real quick.

**Comment by khaneliman** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 365951`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>element-desktop</li>
    <li>element-web (element-web-unwrapped)</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>element-desktop</li>
    <li>element-web (element-web-unwrapped)</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>element-web (element-web-unwrapped)</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>element-web (element-web-unwrapped)</li>
  </ul>
</details>

**Review by khaneliman — APPROVED** [review]
Thanks for cleaning that up! Appreciate it.

**Pull request merged**